### PR TITLE
1439 socket port none

### LINKED ISSUE #1439: Sometimes the method *tcpserver.get_socket_addresses(...)* receives *None* as arguments and crashes
Sometimes the method *get_socket_addresses* receives *None* as arguments and crashes.
Not sure how to reproduce.

```python
ERROR    [golem.client                       ] task_server.sync_network failed
Traceback (most recent call last):
  File "/home/ggruszczynski/PycharmProjects/golem/golem/client.py", line 995, in __do_work
    self.task_server.sync_network()
  File "/home/ggruszczynski/PycharmProjects/golem/golem/task/taskserver.py", line 105, in sync_network
    self.send_waiting_payments()
  File "/home/ggruszczynski/PycharmProjects/golem/golem/task/taskserver.py", line 1012, in send_waiting_payments
    session_cbk=lambda session, payment: session.inform_worker_about_payment(payment)
  File "/home/ggruszczynski/PycharmProjects/golem/golem/task/taskserver.py", line 986, in _send_waiting
    args={'obj': elem}
  File "/home/ggruszczynski/PycharmProjects/golem/golem/network/transport/tcpserver.py", line 155, in _add_pending_request
    self.get_socket_addresses(task_owner, port, key_id) if
  File "/home/ggruszczynski/PycharmProjects/golem/golem/task/taskserver.py", line 512, in get_socket_addresses
    socket_addresses = PendingConnectionsServer.get_socket_addresses(self, node_info, port, key_id)
  File "/home/ggruszczynski/PycharmProjects/golem/golem/network/transport/tcpserver.py", line 231, in get_socket_addresses
    socket_addresses = [SocketAddress(i, port) for i in node_info.prv_addresses]
  File "/home/ggruszczynski/PycharmProjects/golem/golem/network/transport/tcpserver.py", line 231, in <listcomp>
    socket_addresses = [SocketAddress(i, port) for i in node_info.prv_addresses]
  File "/home/ggruszczynski/PycharmProjects/golem/golem/network/transport/tcpnetwork.py", line 56, in __init__
    self.__validate()
  File "/home/ggruszczynski/PycharmProjects/golem/golem/network/transport/tcpnetwork.py", line 68, in __validate
    type(self.port).__name__)
TypeError: Port must be an int, not a NoneType
```

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #1439

### CODE REVIEW

File: golem/task/taskserver.py [hunk 1]
```diff
@@ -1027,6 +1027,10 @@ def _send_waiting(self, elems_set, subtask_id_getter, req_type, session_cbk, p2p
                     logger.debug('Empty node info in %r', elem)
                     elems_set.remove(elem)
                     continue
+                if not isinstance(p2p_node.prv_port, int):
```
Comment (etam) [inline golem/task/taskserver.py:1031]: Do we know why this happens at all? Is it legal for this object to have None as prv_port?

File: tests/golem/task/test_taskserver.py [hunk 2]
```diff
@@ -807,6 +808,20 @@ def test_send_waiting(self, find_sessions_mock, add_pending_mock):
         session_cbk.assert_called_once_with(session, elem)
         session_cbk.reset_mock()
         self.assertEqual(0, len(kwargs['elems_set']))
+        add_pending_mock.reset_mock()
+
+        # Test None instead of a port
```
Comment (etam) [inline tests/golem/task/test_taskserver.py:812]: This test seems too big. I think it should be split, but not necessarily in this PR.

File: golem/task/taskserver.py (hunk 1 above)
Comment (maaktweluit) [reply golem/task/taskserver.py:1031]: from what i've read elsewhere it picks the public port when the private port is not set. i'm not 100% sure it is also the case here.

Comment (jiivan) [reply golem/task/taskserver.py:1031]: Yes `prv_port` can be unknown and this prohibits making `TaskSession` connection.

Comment (etam) [reply golem/task/taskserver.py:1031]: In such case, is it legal to even try to send some pending payments to node that we shouldn't connect to?

Comment (jiivan) [reply golem/task/taskserver.py:1031]: Yes, possibly after restart we can loose information about `prv_port`.

**Comment by maaktweluit** [comment]
Seems this PR is done when its updated and maybe an issue is created for splitting the test

@etam @jiivan is this correct?
Shall i update the conflict?

**Comment by jiivan** [comment]
@maaktweluit correct. Conflict resolved.

File: tests/golem/task/test_taskserver.py [hunk 3]
```diff
@@ -394,7 +394,7 @@ def test_send_waiting_results(self):
 
         ts.failures_to_send[subtask_id] = wtf
         ts.sync_network()
-        ts._add_pending_request.assert_called()
+        self.assertEquals(ts._add_pending_request.call_count, 1)
```
Comment (maaktweluit) [inline tests/golem/task/test_taskserver.py:397]: `assert_called_once()` might be more readable, but same

**Review by maaktweluit — APPROVED** [review]
LGTM, small comment

File: tests/golem/task/test_taskserver.py (hunk 3 above)
Comment (jiivan) [reply tests/golem/task/test_taskserver.py:397]: `assert_called_once()` was added in python3.6 and we still support 3.5 especially in CI.

SEE: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once

Comment (maaktweluit) [reply tests/golem/task/test_taskserver.py:397]: thanks for the info! my CI might break now in this :)

Comment (jiivan) [reply tests/golem/task/test_taskserver.py:397]: It might work when using external mock module, but that should be avoided IMO.

**Pull request merged**